elasticapm/traces.py

```diff
@@ -481,13 +557,63 @@ def end(self, skip_frames=0, duration=None):
         else:
             self.frames = None
         execution_context.set_span(self.parent)
-        tracer.queue_func(SPAN, self.to_dict())
+
+        p = self.parent if self.parent else self.transaction
         if self.transaction._breakdown:
-            p = self.parent if self.parent else self.transaction
-            p.child_ended(self.start_time + self.duration)
+            self._child_durations.stop(self.start_time + self.duration)
             self.transaction.track_span_duration(
                 self.type, self.subtype, self.duration - self._child_durations.duration
             )
+        p.child_ended(self)
+
+    def report(self) -> None:
+        self.tracer.queue_func(SPAN, self.to_dict())
+
+    def try_to_compress(self, sibling: SpanType) -> bool:
```

Comment:
Maybe some docstrings through here so we can remember at a glance what composite vs regular means?